test/test_autograd.py

```diff
@@ -4297,6 +4297,29 @@ def fn2(x):
         check(fast_mode=True)
         check(fast_mode=False)
 
+    def test_gradcheck_forward_ad(self):
```

Comment:
Probably want to test a failure case to catch things like accidental early returns

Reply:
Added with Jeffrey suggestion above to unpack/repack with noisy gradients.